modules/videoio/test/test_gstreamer.cpp

```diff
@@ -151,4 +151,22 @@ TEST(videoio_gstreamer, gray16_writing)
     EXPECT_EQ(0, remove(temp_file.c_str()));
 }
 
+TEST(videoio_gstreamer, timeout_property)
+{
+    if (!videoio_registry::hasBackend(CAP_GSTREAMER))
+        throw SkipTestException("GStreamer backend was not found");
+    VideoCapture cap;
+    cap.open("videotestsrc ! appsink", CAP_GSTREAMER);
+    ASSERT_TRUE(cap.isOpened());
+    const double default_timeout = 30000; // 30 seconds
+    const double open_timeout = 5678; // 3 seconds
+    const double read_timeout = 1234; // 1 second
+    EXPECT_NEAR(default_timeout, cap.get(CAP_PROP_OPEN_TIMEOUT_MSEC), 1e-3);
+    EXPECT_NEAR(default_timeout, cap.get(CAP_PROP_READ_TIMEOUT_MSEC), 1e-3);
```

Comment:
If `cap.get(CAP_PROP_READ_TIMEOUT_MSEC)` is zero then we should bailout or skip this tests (as GStreamer version is very old).

Currently it fails:

```
[ RUN      ] videoio_gstreamer.timeout_property
[ WARN:0@10.679] global cap_gstreamer.cpp:1427 open OpenCV | GStreamer warning: Cannot query video position: status=1, value=0, duration=-1
[ INFO:0@10.679] global cap_gstreamer.cpp:1126 open OpenCV | GStreamer: videotestsrc ! appsink
[ INFO:0@10.679] global cap_gstreamer.cpp:1159 open OpenCV | GStreamer: mode - MANUAL
/build/precommit_custom_linux/4.x/opencv/modules/videoio/test/test_gstreamer.cpp:165: Failure
The difference between default_timeout and cap.get(CAP_PROP_READ_TIMEOUT_MSEC) is 30000, which exceeds 1e-3, where
default_timeout evaluates to 30000,
cap.get(CAP_PROP_READ_TIMEOUT_MSEC) evaluates to 0, and
1e-3 evaluates to 0.001.
[ WARN:0@10.680] global cap_gstreamer.cpp:1772 setProperty OpenCV | GStreamer warning: GStreamer before 1.10 does not support read timeout
/build/precommit_custom_linux/4.x/opencv/modules/videoio/test/test_gstreamer.cpp:169: Failure
The difference between read_timeout and cap.get(CAP_PROP_READ_TIMEOUT_MSEC) is 1234, which exceeds 1e-3, where
read_timeout evaluates to 1234,
cap.get(CAP_PROP_READ_TIMEOUT_MSEC) evaluates to 0, and
1e-3 evaluates to 0.001.
[  FAILED  ] videoio_gstreamer.timeout_property (2 ms)
```